### PR TITLE
installer/*: Add --continue-on-error to destroy workflow 

### PR DESCRIPTION
**Pull request by eparis**
This would let us skip destroying things like worker machines. Useful if the
API never came up. As those resources can't exist yet.

### CODE REVIEW

**Comment by eparis** [comment]
/retest
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/252/pull-ci-openshift-installer-e2e-aws/274 doesn't look like my fault

**Comment by bison** [comment]
Makes sense to me. I'm not an owner here though. You'll need someone else to approve.

**Comment by wking** [comment]
> https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/252/pull-ci-openshift-installer-e2e-aws/274 doesn't look like my fault

Pasting from [here][1], so I can find this later if we see it again:

```
Failure [592.691 seconds]
[BeforeSuite] BeforeSuite 
/tmp/openshift/build-rpms/rpm/BUILD/origin-4.0.0/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/e2e.go:140

  Node 1 disappeared before completing BeforeSuite

  /tmp/openshift/build-rpms/rpm/BUILD/origin-4.0.0/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/e2e.go:140
```

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/252/pull-ci-openshift-installer-e2e-aws/274/build-log.txt

**Comment by wking** [comment]
Same [Node 1 disappeared before completing BeforeSuite][1] this time.  I'll see if I can reproduce in a locally-launched cluster on master.

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/252/pull-ci-openshift-installer-e2e-aws/280/build-log.txt

**Comment by wking** [comment]
Probably won't address the "Node 1 disappeared..." because it only updated the smoke config, but I'm going to kick these again now that openshift/release/pull/1517 has landed:

/test e2e-aws
/test e2e-aws-smoke

**Comment by wking** [comment]
We're back to an earlier release config with openshift/release#1518.  Kicking again:

/retest

**Comment by wking** [comment]
@crawford pointed out that this approach doesn't work well for CI.  We want something like `--keep-going` to not die on in-cluster errors.  Then CI can always set that flag, get full cleanup when the cluster is working, and get out-of-cluster cleanup when the cluster is broken.

**Comment by wking** [comment]
And e2e-aws is [still][1] dying with "Node 1 disappeared before completing BeforeSuite" :/.

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/252/pull-ci-openshift-installer-e2e-aws/283/build-log.txt

**Comment by eparis** [comment]
As I think about it, we should probably make it something like "don't fail" instead of skip. So we can use it in ci? Maybe

**Comment by enxebre** [comment]
For the uninstall story, how about we ping the api health-check before trying to scale down, if API is not responding then just forget about machineset and go ahead with tf destroy
cc @eparis @bison

**Comment by wking** [comment]
> For the uninstall story, how about we ping the api health-check before trying to scale down...

This is racy.  What if the API dies right after you check it?  I'd rather just try and scale down, but keep going if it fails (wking/openshift-installer@945e0a91c0b0fd)

File: installer/cmd/tectonic/main.go [hunk 1]
```diff
@@ -16,8 +16,9 @@ var (
 	clusterInstallCommand = kingpin.Command("install", "Create a new Tectonic cluster")
 	clusterInstallDirFlag = clusterInstallCommand.Flag("dir", "Cluster directory").Default(".").ExistingDir()
 
-	clusterDestroyCommand = kingpin.Command("destroy", "Destroy an existing Tectonic cluster")
-	clusterDestroyDirFlag = clusterDestroyCommand.Flag("dir", "Cluster directory").Default(".").ExistingDir()
+	clusterDestroyCommand   = kingpin.Command("destroy", "Destroy an existing Tectonic cluster")
+	clusterDestroyDirFlag   = clusterDestroyCommand.Flag("dir", "Cluster directory").Default(".").ExistingDir()
+	clusterDestroyContOnErr = clusterDestroyCommand.Flag("continue-on-error", "Log errors, but attempt to continue cleaning up the cluster.  THIS MAY LEAK RESOURCES, because you may not have enough state left after a partial removal to be able to perform a second destroy.").Default("false").Bool()
```
Comment (wking) [inline installer/cmd/tectonic/main.go]: nit: Is there a precedent for `--continue-on-error`?  `--keep-going` is [in Make][1].  Consistency with existing tools isn't critical, but it's nice to reuse existing wording where we can.

[1]: https://www.gnu.org/software/make/manual/html_node/Options-Summary.html#index-_002d_002dkeep_002dgoing-2

Comment (eparis) [reply installer/cmd/tectonic/main.go]: I was thinking of [spf13/pflag](https://github.com/spf13/pflag/blob/master/flag.go#L120) which is, admittedly, obscure.

**Comment by wking** [comment]
/hold cancel

I'm fine with 23915f2e7e as it stands, or [shifted back to `--keep-going][1], either way.  Does anyone else want to take a look before I `/lgtm`?

[1]: https://github.com/openshift/installer/pull/252#discussion_r218152276

**Comment by crawford** [comment]
@wking Looking

**Pull request merged**